### ISSUE TITLE
Test failures due to Mockito calling real methods of mocked objects (Zing, IBM-8, OpenJ-9) [HZ-2245]

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast#23477]
_master_ (commit 006b3f3c9878902973b57adca9dcbab0c6e8bdc2)

Failed on zing-11: 
- https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/10/testReport/com.hazelcast.flakeidgen.impl/FlakeIdGeneratorProxyTest/
- https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/10/testReport/com.hazelcast.internal.metrics.impl/MetricsServiceTest
- https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/10/testReport/com.hazelcast.internal.partition.impl/PartitionReplicaStateChecker_triggerAndWaitForReplicaSyncTest/
- https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/10/testReport/com.hazelcast.map.impl.record/DataRecordFactoryTest/
- https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/10/testReport/com.hazelcast.map.impl.record/ObjectRecordFactoryTest/
- https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/10/testReport/com.hazelcast.query.impl/IndexImplTest/
- https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/10/testReport/com.hazelcast.replicatedmap.impl.record/LazyCollectionTest/

Almost all tests in following test classes failed with various NPE:
- `com.hazelcast.flakeidgen.impl.FlakeIdGeneratorProxyTest`
- `com.hazelcast.internal.metrics.impl.MetricsServiceTest`
- `com.hazelcast.internal.partition.impl.PartitionReplicaStateChecker_triggerAndWaitForReplicaSyncTest`
- `com.hazelcast.map.impl.record.DataRecordFactoryTest`
- `com.hazelcast.map.impl.record.ObjectRecordFactoryTest`
- `com.hazelcast.query.impl.IndexImplTest`
- `com.hazelcast.replicatedmap.impl.record.LazyCollectionTest`

<details><summary>Stacktrace for com.hazelcast.flakeidgen.impl.FlakeIdGeneratorProxyTest:</summary>

```
java.lang.NullPointerException
	at com.hazelcast.flakeidgen.impl.FlakeIdGeneratorService.getLocalFlakeIdStats(FlakeIdGeneratorService.java:108)
	at com.hazelcast.flakeidgen.impl.FlakeIdGeneratorService.updateStatsForBatch(FlakeIdGeneratorService.java:101)
	at com.hazelcast.flakeidgen.impl.FlakeIdGeneratorProxy.newIdBaseLocal(FlakeIdGeneratorProxy.java:190)
	at com.hazelcast.flakeidgen.impl.FlakeIdGeneratorProxyTest.when_twoIdsAtTheSameMoment_then_higherSeq(FlakeIdGeneratorProxyTest.java:203)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:887)

```
</details>

<details><summary>Stacktrace for com.hazelcast.internal.metrics.impl.MetricsServiceTest:</summary>

```
java.lang.NullPointerException
	at com.hazelcast.spi.impl.NodeEngineImpl.getConfig(NodeEngineImpl.java:302)
	at com.hazelcast.spi.impl.executionservice.impl.ExecutionServiceImpl.register(ExecutionServiceImpl.java:175)
	at com.hazelcast.spi.impl.executionservice.impl.ExecutionServiceImpl.register(ExecutionServiceImpl.java:163)
	at com.hazelcast.spi.impl.executionservice.impl.ExecutionServiceImpl.registerExecutors(ExecutionServiceImpl.java:149)
	at com.hazelcast.spi.impl.executionservice.impl.ExecutionServiceImpl.<init>(ExecutionServiceImpl.java:134)
	at com.hazelcast.internal.metrics.impl.MetricsServiceTest.setUp(MetricsServiceTest.java:117)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:365)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:258)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:41)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)

```
</details>

<details><summary>Stacktrace for com.hazelcast.internal.partition.impl.PartitionReplicaStateChecker_triggerAndWaitForReplicaSyncTest:</summary>

```
java.lang.NullPointerException
	at com.hazelcast.internal.partition.impl.PartitionReplicaStateChecker.getPartitionTableState(PartitionReplicaStateChecker.java:104)
	at com.hazelcast.internal.partition.impl.PartitionReplicaStateChecker.getPartitionServiceState(PartitionReplicaStateChecker.java:83)
	at com.hazelcast.internal.partition.impl.PartitionReplicaStateChecker_triggerAndWaitForReplicaSyncTest.whenCheckAndTriggerReplicaSync_withNeedsReplicaStateCheck_thenReturnTrue(PartitionReplicaStateChecker_triggerAndWaitForReplicaSyncTest.java:157)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:887)

```
</details>

<details><summary>Stacktrace for com.hazelcast.map.impl.record.DataRecordFactoryTest:</summary>

```
java.lang.NullPointerException
	at com.hazelcast.map.impl.record.DataRecordFactory.newRecord(DataRecordFactory.java:41)
	at com.hazelcast.map.impl.record.AbstractRecordFactoryTest.test_expected_record_per_config_is_created(AbstractRecordFactoryTest.java:67)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:887)

```
</details>

<details><summary>Stacktrace for com.hazelcast.map.impl.record.ObjectRecordFactoryTest:</summary>

```
java.lang.NullPointerException
	at com.hazelcast.map.impl.record.ObjectRecordFactory.newRecord(ObjectRecordFactory.java:43)
	at com.hazelcast.map.impl.record.AbstractRecordFactoryTest.test_expected_record_per_config_is_created(AbstractRecordFactoryTest.java:67)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:887)

```
</details>

<details><summary>Stacktrace for com.hazelcast.query.impl.IndexImplTest:</summary>

```
java.lang.NullPointerException
	at com.hazelcast.query.impl.QueryableEntry.extractAttributeValue(QueryableEntry.java:116)
	at com.hazelcast.query.impl.QueryableEntry.getAttributeValue(QueryableEntry.java:62)
	at com.hazelcast.query.impl.AbstractIndex.extractAttributeValue(AbstractIndex.java:354)
	at com.hazelcast.query.impl.AbstractIndex.putEntry(AbstractIndex.java:132)
	at com.hazelcast.query.impl.IndexImplTest.saveEntryIndex_doNotDeserializeKey(IndexImplTest.java:69)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:887)

```
</details>

<details><summary>Stacktrace for com.hazelcast.replicatedmap.impl.record.LazyCollectionTest:</summary>

```
java.lang.NullPointerException
	at com.hazelcast.replicatedmap.impl.record.InternalReplicatedMapStorage.values(InternalReplicatedMapStorage.java:85)
	at com.hazelcast.replicatedmap.impl.record.LazyCollection.<init>(LazyCollection.java:36)
	at com.hazelcast.replicatedmap.impl.record.LazyCollectionTest.setUp(LazyCollectionTest.java:43)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:365)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at com.hazelcast.test.HazelcastParallelClassRunner.access$301(HazelcastParallelClassRunner.java:49)
	at com.hazelcast.test.HazelcastParallelClassRunner$TestRunner.run(HazelcastParallelClassRunner.java:162)
	at java.base/java.lang.Thread.run(Thread.java:887)

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-ZingJDK11/10/

**Comment by sumnerib** [comment]
Failed again on _master_ with zing-11: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/11/

**Comment by olukas** [comment]
Failed again on master with:
- zing-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK17/12/
- zing-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/483/

**Comment by sumnerib** [comment]
Failed again on `master` with zing-11: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/14/

**Comment by olukas** [comment]
Failed again on master with:
- zing-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/485/
- zing-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK17/14/

**Comment by Patras3** [comment]
Failed again on master with:
* zing-17: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK17/15/

**Comment by sumnerib** [comment]
Failed again on _master_ with zing-8: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK8/487/

**Comment by olukas** [comment]
Failed again on master with
- ZingJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/488/

**Comment by Patras3** [comment]
Failed again on _master_ with zing-17: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK17/18/

**Comment by arodionov** [comment]
It seems that the issue is related to ByteBuddy code generation for mocked methods (`InlineByteBuddyMockMaker#createMock`).  Problematic JDK platforms: Zing (Azul), Ibm

**Comment by olukas** [comment]
Failed again on master with:
- zing-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/490/
- zing-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK17/19/

**Comment by ramizdundar** [comment]
@arodionov https://github.com/hazelcast/hazelcast-enterprise/issues/5661 is also related to this.

**Comment by ramizdundar** [comment]
Maybe we could rename the issue as: `Test failures due to Mockito calling real methods of mocked objects`?

**Comment by arodionov** [comment]
The first TF related to mocks happened ~ on Jan 20.
 On Jan 19 we updated the mockito-core from 3.6.0 to 4.11.0 https://github.com/hazelcast/hazelcast/pull/23277

A possible issue in Mockito is https://github.com/mockito/mockito/pull/2812

**Comment by arodionov** [comment]
Possible causes of the issue:
- Mockinto issue https://github.com/mockito/mockito/pull/2812 
- Byte Buddy Java agent mock maker - agent does not connect correctly and should be attached manually (less possible because there is no errors in logs https://javadoc.io/static/net.bytebuddy/byte-buddy-agent/1.12.8/net/bytebuddy/agent/ByteBuddyAgent.html#install--)

**Comment by mdumandag** [comment]
One more possible related test failure: 
- https://github.com/hazelcast/hazelcast/issues/23468

See: https://github.com/hazelcast/hazelcast/issues/23468#issuecomment-1496027540

**Comment by olukas** [comment]
Reopen since it seems to behave the same on ZingJDK. Failed again on master with:
- ZingJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK17/21/
- ZingJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/492/

**Comment by Patras3** [comment]
Reopening, somehow sync of my fork, closed this issue.

**Comment by olukas** [comment]
Failed again on master with:
- ZingJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK17/22/
- ZingJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/

**Comment by olukas** [comment]
Failed again on master with:
- ZingJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK17/23/
- ZingJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/494/

**Comment by ramizdundar** [comment]
@arodionov I think these 2 issues are also related, so maybe you can update your [list](https://github.com/hazelcast/hazelcast/issues/23477#issuecomment-1489878449) if you also think so:

- https://github.com/hazelcast/hazelcast/issues/24287
- https://github.com/hazelcast/hazelcast/issues/24274 - this one is interesting because it's openjdk19

**Comment by olukas** [comment]
Failed again on master with:
- ZingJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/495/
- ZingJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK17/24/

**Comment by olukas** [comment]
Adding `constant-test-failure` label since it fails constantly on Zing.

**Comment by JamesHazelcast** [comment]
Once the following PRs are merged, please report any new test failures so we can see if Mockito's update provides us any improvements.

https://github.com/hazelcast/hazelcast/pull/24889
https://github.com/hazelcast/hazelcast-enterprise/pull/6179

**Comment by Patras3** [comment]
Failed again on _master_ with zing-17:
Example: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK17/32/testReport/com.hazelcast.flakeidgen.impl/FlakeIdGeneratorProxyTest/test_negativeId/

**Comment by sumnerib** [comment]
Failed again on master with zing-17: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK17/33/

**Comment by sumnerib** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/job/Hazelcast-master-Openj9-11/141/

**Comment by sumnerib** [comment]
Failed again on _master_ with zing-17: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK17/37/

**Comment by sumnerib** [comment]
Failed again on _5.3.z_ with zing-8: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-ZingJDK8/153/

**Comment by sumnerib** [comment]
Failed again on _master_ with zing-17: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK17/39/

**Comment by JamesHazelcast** [comment]
Making a note here: Azul Systems (Zing/Prime JDK creators) have confirmed these Mockito-related failures are reproducible for them as well across all JDKS of Zing, and they are working on a fix for us.

**Comment by Patras3** [comment]
@JamesHazelcast any updates on this? Is there any way to track this external issue?

**Comment by JamesHazelcast** [comment]
@Patras3 I've posted updates on the associated Jira: https://hazelcast.atlassian.net/browse/HZ-2245

TL;DR, I don't think we can expect a fix from their side anytime soon, they themselves seem to suggest we don't use their JDK outside of production (see my Jira comment for details).

**Comment by olukas** [comment]
@JamesHazelcast  Should not we ignore those tests in our code for Zing JDK in that case? (I don't see any reason to have a tests like this which are expected to fail)

**Comment by JamesHazelcast** [comment]
@olukas they're flakey, but not exactly expected to fail - they fail periodically under load, but not consistently.

**Comment by sumnerib** [comment]
Failed again on _master_ with zing-21: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK21/4/

**Comment by AleksPeychev** [comment]
Failed again on _5.4.z_ with zing-21: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-ZingJDK21/6/testReport/junit/com.hazelcast.internal.partition.impl/PartitionReplicaStateChecker_triggerAndWaitForReplicaSyncTest/

**Comment by olukas** [comment]
Closing the test failure - it hasn't failed in the last 3 months.
